doc/adr/0001-use-micro-frontends.md

```diff
@@ -0,0 +1,31 @@
+# 1. Use micro frontends
+
+Date: 2020-03-25
+
+## Status
+
+Proposed
+
+## Context
+
+We have designed the API layer (known as the backend) to be a custom set of services. This allows the user to tailor it
+to their own requirements, and introduce custom code as needed.
+
+We are currently planning to make the presentation layer (known as the frontend) a monolith. That is, a single service
+that generates all the HTML pages.
+
+As a monolith, it is hard, if not impossible, to change or replace parts without having to replace the whole. This will
+mean a lot of configuration options to support differing requirements. It will also limit adoption of the platform as it
+can't meet them.
+
+## Decision
+
+We will create separate services for individual, and groups of, pages, that will be exposed through a proxy to appear as
+a single service. The frontends are entirely independent, and can be not used or replaced entirely at will.
+
+## Consequences
+
+- A proxy (e.g. Nginx, Varnish) is required to route between the frontends.
+- Authentication knowledge will need to be shared between the frontends.
+- Each frontend will have its own assets.
+- Each frontend can be configured to know how to generate links to other frontends.
```

Comment:
Given this, a proxy may be suggested rather than required, if the links could be configured to go to different origins (e.g. hostnames or ports). Not that I am suggesting to support that.
Or is a single hostname required to, for example, share a JWT cookie?

Reply:
Cookies would probably need `Path=/; Domain=.example.com`, so the domain name needs to be the same at least.